code/modules/cargo/materials_market.dm

```diff
@@ -149,62 +148,71 @@
 					break
 			if(!material_bought)
 				CRASH("Invalid material name passed to materials market!")
-			var/mob/living/living_user = usr
-			var/datum/bank_account/account_payable = SSeconomy.get_dep_account(ACCOUNT_CAR)
-			if(ordering_private)
-				var/obj/item/card/id/used_id_card = living_user.get_idcard(TRUE)
+			var/mob/living/living_user = ui.user
+			var/obj/item/card/id/used_id_card = living_user.get_idcard(TRUE)
+			if(isnull(used_id_card))
+				say("No ID Found")
+				return
```

Comment:
Minor thing, call this at the start of the switch case since there's in all likelyhood a higher chance of a user not having an ID card than the material being invalid for the purposes of early returns.

Reply:
Done